tensorlayer/distributed.py

```diff
@@ -218,17 +209,8 @@ def train_on_batch(self):
         """Train a mini-batch."""
         self._sess.run(self._train_op)
 
-    def train_to_end(self):
```

Comment:
why train to end is removed?

Reply:
`train_to_end` is actually `train_on_batch` with a while loop. This functionality can be easily implemented and also modified by users. In order to allow more flexibility for users, we may need to remove `train_to_end`.

Reply:
After talking to Jingqing yesterday, now I have a much more clear view about the trainer abstraction. In principle, we shall keep the trainer interface very flexible. For this purpose, we shall actually expose the internal session to the users and let them manipulate them. In addition we shall add a bunch of helper functions to help them validate the training error and adapt hyper-parameters based on epoch, instead of the global_steps which are very confused in a parallel setting. 

@zsdonghao @JingqingZ